package/kernel/linux/modules/crypto.mk

```diff
@@ -75,6 +75,28 @@ endef
 $(eval $(call KernelPackage,crypto-authenc))
 
 
+define KernelPackage/crypto-blake2s
+  TITLE:=BLAKE2s digest algorithm CryptoAPI module
+  KCONFIG:= \
+	CONFIG_CRYPTO_BLAKE2S \
```

Comment:
WireGuard only uses the library code.

Actually, wireguard should automatically select all the library bits that it needs, and none of those modules then need to be autoloaded, as the wireguard module depends directly on them.

Reply:
I guess it's a tradeoff of who else will use these vs what is the downside of building modules+libs together and autoloading? Module size, performance? I took a stab at it here, but better ideas are welcome - these crypto modules are a little complex with modules+libs+hw-specific stuff, so better ideas are welcome. Or, we can start here and let the community improve it as needed.

Reply:
The cryptoapi module is for the plugable crypto stuff that wireguard doesn't use. So this winds up amounting to simply having more code and disk space for something very unlikely to be used.

Reply:
Okay - I'll take a stab at taking that out. wireguard is the only user of these here at the moment.

Reply:
Okay - should be quite a bit leaner now - please take a look.